ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -179,12 +179,12 @@ void startReadingFromChannel() {
       public void onCompletion(Long result, Exception exception) {
         if (exception != null) {
           setOperationExceptionAndComplete(exception);
+          routerCallback.onPollReady();
         } else {
           blobSize = result;
           chunkFillingCompletedSuccessfully = true;
         }
         chunkFillerChannel.close();
-        routerCallback.onPollReady();
```

Comment:
That is correct. Things become actionable in a subsequent `poll()` of the `PutOperation` only when in response to the channel getting closed here, the `ChunkFillerThread` finishes off and makes the last chunk `Ready`, so that is the correct place to invoke `onPollReady()` (unless the channel closes with an error, in which case the `ChunkFillerThread` does nothing, so the callback is invoked here in the error path).